tests/unit/Makefile

```diff
@@ -0,0 +1,43 @@
+GTEST_DIR = ./googletest/googletest
+
+CODEDIR = ../..
+EXECUTABLE = tests
+
+# Where to find user code.
+CPPFLAGS += -isystem $(GTEST_DIR)/include -std=c++11
+CXXFLAGS += -g -ldl -pthread
+
+TESTS = test_events.cc test_virtclk.cc
+OBJECTS=$(TESTS:.cc=.o)
+
+GTEST_HEADERS := $(GTEST_DIR)/include/gtest/*.h \
+                $(GTEST_DIR)/include/gtest/internal/*.h
+
+all : googletest code $(OBJECTS) gtest_main.a 
+	$(CXX) $(wildcard $(CODEDIR)/*.o) $(OBJECTS) gtest_main.a $(CPPFLAGS) $(CXXFLAGS) -o $(EXECUTABLE)
+
+code :
+	make -C $(CODEDIR)
+
+googletest:
+	git clone git@asof.atl.calpoly.edu:libraries/googletest
```

Comment:
Why do you have a submodule and a git clone command? Also, since this is a public project we should make a github fork of google test so external users can pull from the repo.